mne/io/edf/edf.py

```diff
@@ -396,7 +421,21 @@ def _get_info(fname, stim_channel, eog, misc, exclude, preload):
         chan_info['coord_frame'] = FIFF.FIFFV_COORD_HEAD
         chan_info['coil_type'] = FIFF.FIFFV_COIL_EEG
         chan_info['kind'] = FIFF.FIFFV_EEG_CH
-        chan_info['loc'] = np.zeros(12)
+        chan_info['loc'] = nan_arr
```

Comment:
This needed to be `nan_arr.copy()` (as found with @drammock working together), after this PR all `info['chs'][ii]['loc']` point to the same array in memory, so setting values in one of them sets all of them. We need a unit test that would have caught this...

Reply:
(I think @drammock is working on a fix for this)

Reply:
see #9743